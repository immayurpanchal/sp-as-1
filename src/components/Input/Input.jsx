import React, { useState } from 'react';

const Input = (props) => {
	const { label, id, value, validate, errorMessage, ...restProps } = props;

	const [isValid, setValid] = useState(false);
	const [isTouched, setTouched] = useState(false);

	const getStyle = () => {
		if (isTouched) {
			return isValid ? 'is-valid' : 'is-invalid';
		}
		return '';
	};

	return (
		<>
			{label && (
				<label htmlFor={id} className='form-label'>
					{label}
				</label>
			)}
			<input
				{...restProps}
				name={id}
				className={`form-control ${getStyle()}`}
				onFocus={(e) => {
					typeof validate === 'function' && setValid(props.validate(e));
					setTouched(true);
				}}
				onChange={(e) => {
					typeof validate === 'function' && setValid(props.validate(e));
					props.onChange(e);
				}}
				onBlur={(e) => {
					typeof validate === 'function' && setValid(props.validate(e));
					setTouched(true);
				}}
			/>
			<div className='invalid-feedback'>{errorMessage || ''}</div>
		</>
	);
};

export default Input;
