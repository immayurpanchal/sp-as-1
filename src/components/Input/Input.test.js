import { render } from '@testing-library/react'
import Input from '.'

// describe is used to define the test suite.
describe('Input component', () => {
  // for individual scenario testing, use it()
  // this way the test will written in a declarative way.
  // it will be easy to read and maintain
  it('render input component', () => {
    const tree = render(
      <Input
        key='bedroom'
        required
        errorMessage='Bedroom must be between 1 and 10'
        id='bedroom'
        label='Bedroom'
        max='10'
        min='1'
        type='number'
        validate={({ target: { value } }) => +value >= 1 && +value <= 10}
        value={0}
        onChange={() => {
          // Do nothing
        }}
      />
    )

    // Example of Snapshot Testing
    expect(tree).toMatchSnapshot()
  })
})
