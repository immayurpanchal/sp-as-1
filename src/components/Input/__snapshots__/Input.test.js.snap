// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component render input component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="col-sm-2 col-form-label"
        for="bedroom"
      >
        Bedroom
      </label>
      <div
        class="col-sm-10"
      >
        <input
          class="form-control"
          max="10"
          min="1"
          name="bedroom"
          required=""
          type="number"
          value=""
        />
        <div
          class="invalid-feedback"
        >
          Bedroom must be between 1 and 10
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <label
      class="col-sm-2 col-form-label"
      for="bedroom"
    >
      Bedroom
    </label>
    <div
      class="col-sm-10"
    >
      <input
        class="form-control"
        max="10"
        min="1"
        name="bedroom"
        required=""
        type="number"
        value=""
      />
      <div
        class="invalid-feedback"
      >
        Bedroom must be between 1 and 10
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
